R/tm_a_pca.R

```diff
@@ -876,20 +877,50 @@ srv_a_pca <- function(id, data, reporter, filter_panel_api, dat, plot_height, pl
     # plot final ----
     output_q <- reactive({
       req(computation())
-      if (input$plot_type == "Elbow plot") {
-        plot_elbow(computation())
-      } else if (input$plot_type == "Circle plot") {
-        plot_circle(computation())
-      } else if (input$plot_type == "Biplot") {
-        plot_biplot(computation())
-      } else if (input$plot_type == "Eigenvector plot") {
-        plot_pc_var(computation())
-      } else {
-        stop("Unknown plot")
-      }
+      iv_extra <- shinyvalidate::InputValidator$new()
+      iv_extra$add_rule("ggtheme", shinyvalidate::sv_required("Please select a theme."))
+
+      iv_elbow <- shinyvalidate::InputValidator$new()
+      iv_elbow$condition(~ input$plot_type == "Elbow plot")
```

Comment:
As you are doing this inside the reactive then you already know what input$plot_type is so you only need to create the one validator for that specific type rather than creating 4